### ISSUE TITLE
windowrule( workspace id) can't apply to the application running in xwayland.

### ISSUE THREAD

**Issue by command-z-z** [hyprwm/Hyprland#6203]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 7ad9116de8d@b7dac27eaf080ebd92998a8fbue5 dirty ([gha] Nix: update inputs).
Date: Thu May 23 06:43:47 2024
Tag: v@.40.0-122-g7ad9116d, commits: 4728

flags: (if any)

System Information:

System name: Linux

Node name: eugeneArch

Release: 6.9.1-arch1-1

Version: #1 SMP PREEMPT_DYNAMIC Fri, 17 May 2024 16:56:38 +0000

GPU information:
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation AD104 [GeForce RTX 4070] [10de:2786] (rev a1) (prog-if 00 [VGA controller])

os-release: NAME="Arch Linux"

PRETTY_NAME="Arch Linux"

ID=arch

BUILD_ID=rolling

ANSI_COLOR="38 ; 2; 23; 147; 209"

HOME_URL="https://archlinux.org/"

DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo

plugins:
```

</details>

### Bug or Regression?

Bug

### Description

i set the config as follow
```
windowrule  = workspace 2 , (chrome)$
windowrule  = workspace 4 , (Spotify)$
```
it works in native wayland application(e.g., kitty, zathura)

### How to reproduce

shortcut key launch application, the app will appear in the current workspace, but can't be assigned to specify workspace.

**Comment by Agent00Ming** [comment]
https://github.com/hyprwm/Hyprland/issues/6081#issuecomment-2112470404 + will be fixed by #6086

**Comment by command-z-z** [comment]
> Hi command-z-z,
> 
> I got it to work with using v2 and either initialClass or initialTitle, here is my config. Both VS Code and YT Music are xwayland.
> 
> ```
> windowrulev2 = workspace 1, class:^(kitty)$
> windowrulev2 = workspace 2, class:^(firefox)$
> windowrulev2 = workspace 3, initialTitle:^(Visual Studio Code)$
> windowrulev2 = workspace 4, class:^(pcmanfm)$
> windowrulev2 = workspace 5, initialClass:^(YouTube Music)$
> ```
> 
> oh, and I also made these the first rules applied.
> 
> Hope that helps. Lee

sorry, it's not work for me.

**Comment by command-z-z** [comment]
fixed by #6086
